deviceplugin/deviceplugin.go

```
@@ -0,0 +1,144 @@
+// Package deviceplugin provides a basic implementation of a Kubernetes DevicePlugin without content.
+// It is intended to be used to build other DevicePlugins.
+package deviceplugin
+
+import (
+	"log"
+	"net"
+	"os"
+	//      "strings"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1"
+	"path"
+	"time"
+)
+
+type DevicePlugin struct {
+	socket       string
+	resourceName string
+	stop         chan interface{}
+	server       *grpc.Server
+}
+
+func NewDevicePlugin(serversock string, resourcename string) *DevicePlugin {
+	return &DevicePlugin{
+		socket:       serversock,
+		stop:         make(chan interface{}),
+		resourceName: resourcename,
+	}
+}
+
+func dial(timeoutContext context.Context, unixSocketPath string) (*grpc.ClientConn, error) {
+	return grpc.DialContext(
+		timeoutContext,
+		unixSocketPath,
+		grpc.WithInsecure(), grpc.WithBlock(),
+		grpc.WithDialer(func(addr string, timeout time.Duration) (net.Conn, error) {
+			return net.DialTimeout("unix", addr, timeout)
+		}),
+	)
+}
+
+func (d *DevicePlugin) cleanup() error {
+	if err := os.Remove(d.socket); err != nil && !os.IsNotExist(err) {
```

Is there any reason to suppress other errors?